gadget/gadget.go

```diff
@@ -1013,3 +1013,36 @@ func PositionedVolumeFromGadget(gadgetRoot string) (*LaidOutVolume, error) {
 	}
 	return nil, fmt.Errorf("internal error in PositionedVolumeFromGadget: this line cannot be reached")
 }
+
+func flatten(path string, cfg interface{}, out map[string]interface{}) {
+	if cfgMap, ok := cfg.(map[string]interface{}); ok {
+		for k, v := range cfgMap {
+			p := k
+			if path != "" {
+				p = path + "." + k
+			}
+			flatten(p, v, out)
+		}
+	} else {
+		out[path] = cfg
+	}
+}
+
+// CoreDefaults returns default system (or core) configuration from gadget defaults.
+func CoreDefaults(gadgetDefaults map[string]map[string]interface{}, coreSnapID string) (map[string]interface{}, error) {
+	coreDefaults := map[string]interface{}{}
+	if defaults, ok := gadgetDefaults["system"]; ok {
+		flatten("", defaults, coreDefaults)
+		return coreDefaults, nil
+	}
+	if defaults, ok := gadgetDefaults[coreSnapID]; ok {
+		if values, ok := defaults["core"]; ok {
```

Comment:
it this level actualyl there, I'm confused,  I thought the config values would just be under the snapid?

Reply:
Yes, you're right, I got confused myself with this. Will fix. Thanks!

Reply:
see configstate.ConfigDefaults